### PR TITLE
chore: add 404 page for project/pipeline detail

### PR DESCRIPTION
**Pull request by tyyzqmf**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [ ] all code changes are covered by unit tests
- [x] end-to-end tests
  - [x] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting
  - [ ] streaming ingestion
    - [ ] with Redshift Serverless
    - [ ] with provisioned Redshift

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @tyyzqmf on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.80|< 80|
|Duplicated lines density|:white_check_mark: OK|8.60|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|9|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1532)
###### _updated: 7/12/2024, 09:56:55 (UTC+0)_

**Comment by zxkane** [comment]
# SonarQube Code Analytics
## Quality Gate passed

passed

## Additional information
*The following metrics might not affect the Quality Gate status but improving them will improve your project code quality.*

## Issues
bug   [0 Bugs](http://sonarqube:9000/project/issues?id=pr-1532&resolved=false&sinceLeakPeriod=true&types=BUG)

vulnerability   [0 Vulnerabilities](http://sonarqube:9000/project/issues?id=pr-1532&resolved=false&sinceLeakPeriod=true&types=VULNERABILITY)

code_smell   [0 Code Smells](http://sonarqube:9000/project/issues?id=pr-1532&resolved=false&sinceLeakPeriod=true&types=CODE_SMELL)

## Coverage and Duplications
*No data* **Coverage**

*No data* **Duplication**

File: frontend/public/locales/zh-CN/common.json [hunk 1]
```diff
@@ -330,5 +330,9 @@
   "enabled": "已启用",
   "disabled": "已禁用",
   "maxMindCopyRight": "IP 富化插件使用 MaxMind 创建的 Geolite2 数据，可从以下网址获得 ",
-  "checkingAvailable": "检查亚马逊云科技服务的可用性"
+  "checkingAvailable": "检查亚马逊云科技服务的可用性",
+  "error": {
+    "notFoundTitle": "资源不存在",
```
Comment (chenhaiyun) [inline frontend/public/locales/zh-CN/common.json:335]: 页面不存在？

File: frontend/src/pages/projects/detail/ProjectDetail.tsx [hunk 2]
```diff
@@ -56,20 +56,22 @@ const ProjectDetail: React.FC = () => {
...
-      const { success, data }: ApiResponse<IProject> = await getProjectDetail({
+      const res = await getProjectDetail({
         id: projectId,
       });
-      if (success) {
-        setProjectInfo(data);
-        if (data?.pipelineId && data?.pipelineId !== '') {
+      if (res?.success) {
+        setProjectInfo(res?.data);
+        if (res?.data?.pipelineId && res?.data?.pipelineId !== '') {
```
Comment (chenhaiyun) [inline frontend/src/pages/projects/detail/ProjectDetail.tsx:64]: **res?.data?.pipelineId** is enough?

File: frontend/src/pages/pipelines/detail/PipelineDetail.tsx [hunk 3]
```diff
@@ -66,8 +66,10 @@ const PipelineDetail: React.FC = () => {
         setLoadingData(false);
         setLoadingPipeline(false);
       }
-    } catch (error) {
-      setLoadingPipeline(false);
+    } catch (error: any) {
+      if (error?.message?.response?.status === 404) {
+        window.location.href = '/error/404';
```
Comment (zxkane) [inline frontend/src/pages/pipelines/detail/PipelineDetail.tsx:71]: Is the url path `/error/404` a convention in React?

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @tyyzqmf on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.70|< 80|
|Duplicated lines density|:white_check_mark: OK|8.60|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|9|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1532)
###### _updated: 7/15/2024, 03:06:02 (UTC+0)_

File: frontend/src/pages/pipelines/detail/PipelineDetail.tsx (hunk 3 above)
Comment (tyyzqmf) [reply frontend/src/pages/pipelines/detail/PipelineDetail.tsx:71]: There is no such convention in React.

File: frontend/src/pages/error-page/404.tsx
Comment (chenhaiyun) [inline frontend/src/pages/error-page/404.tsx:1]: Update the file name to `NotFound.tsx`

**Review by zxkane — CHANGES_REQUESTED** [review]
Let's not redirect to `/error/404` if accessing a non-existing or deleted project. Rendering the not-found component if the project does not exist.

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @tyyzqmf on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.70|< 80|
|Duplicated lines density|:white_check_mark: OK|8.50|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|9|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1532)
###### _updated: 7/16/2024, 03:12:53 (UTC+0)_

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @tyyzqmf on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.70|< 80|
|Duplicated lines density|:white_check_mark: OK|8.50|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|9|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1532)
###### _updated: 7/16/2024, 03:13:56 (UTC+0)_

File: frontend/public/locales/zh-CN/common.json [hunk 4]
```diff
@@ -330,5 +330,13 @@
...
   "disabled": "已禁用",
   "maxMindCopyRight": "IP 富化插件使用 MaxMind 创建的 Geolite2 数据，可从以下网址获得 ",
-  "checkingAvailable": "检查亚马逊云科技服务的可用性"
+  "checkingAvailable": "检查亚马逊云科技服务的可用性",
+  "error": {
+    "notFoundTitle": "页面不存在",
+    "notFoundMessage": "您要查找的页面不存在。",
+    "notFoundProjectTitle": "项目不存在",
+    "notFoundProjectMessage": "您要查找的页面不存在。",
+    "notFoundPipelineTitle": "流水线不存在",
```
Comment (zxkane) [inline frontend/public/locales/zh-CN/common.json:339]: change to `管道` to match the terminology in documentation

**Review by zxkane — CHANGES_REQUESTED** [review]

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @tyyzqmf on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.70|< 80|
|Duplicated lines density|:white_check_mark: OK|8.50|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|9|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1532)
###### _updated: 7/16/2024, 05:32:39 (UTC+0)_

**Comment by tyyzqmf** [comment]
> Let's not redirect to `/error/404` if accessing a non-existing or deleted project. Rendering the not-found component if the project does not exist.

done.

File: frontend/src/pages/pipelines/detail/comps/NoFoundPipeline.tsx
Comment (chenhaiyun) [inline frontend/src/pages/pipelines/detail/comps/NoFoundPipeline.tsx:1]: We can re-use `NotFound.tsx` component, add **notFoundTitle** and **notFoundDescription** props.

File: frontend/src/pages/projects/detail/comp/NoFoundProject.tsx
Comment (chenhaiyun) [inline frontend/src/pages/projects/detail/comp/NoFoundProject.tsx:1]: re-use `NotFound.tsx`

File: frontend/src/pages/pipelines/detail/comps/NoFoundPipeline.tsx
Comment (tyyzqmf) [reply frontend/src/pages/pipelines/detail/comps/NoFoundPipeline.tsx:1]: done.

File: frontend/src/pages/projects/detail/comp/NoFoundProject.tsx
Comment (tyyzqmf) [reply frontend/src/pages/projects/detail/comp/NoFoundProject.tsx:1]: done.

**Review by chenhaiyun — APPROVED** [review]

**Review by zxkane — APPROVED** [review]

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @tyyzqmf on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.70|< 80|
|Duplicated lines density|:white_check_mark: OK|8.60|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|9|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1532)
###### _updated: 7/17/2024, 02:43:55 (UTC+0)_

**Pull request merged**